src/Locale/LocaleManager.php

```diff
@@ -112,11 +112,4 @@ public function setTranslator(Translator $translator)
     {
         $this->translator = $translator;
     }
-
-    public function clearCache()
```

Comment:
We can't delete this method, it is used elsewhere, `LanguagePack` extender lifecycle hooks for example.

Reply:
Wow! I'm so curious about the sensibility of that. Enabling any extension flushes the whole cache anyway right? Wait I can't find where we do that. So if we don't clear the full cache, wouldn't it make sense to do that because the JS is still cache without clearing that?

Reply:
Enabling or disabling an extension does indirectly clear the full cache, most of that is done through the extenders lifecycle hooks (`LanguagePack`, `Locales`, `Formatter`, `View`) with the exception of frontend assets (js, css) those are cleared/recompiled by listening to the (`Enabled`, `Disabled`, `ClearingCache`) events.

Reply:
So:

ForumServiceProvider and AdminServiceProvider:

```php
        $events->listen(
            [Enabled::class, Disabled::class, ClearingCache::class],
            function () {
                $recompile = new RecompileFrontendAssets(
                    $this->container->make('flarum.assets.forum'),
                    $this->container->make(LocaleManager::class)
                );
                $recompile->flush();
            }
        );
```

Seems to flush only css and js. Not the locale cache.

The Frontend Extender does something similar but for different frontends if they haven't been registered yet (ps this is code duplication, maybe all three locations need a trait?)

```php
            $events->listen(
                [Enabled::class, Disabled::class, ClearingCache::class],
                function () use ($container, $abstract) {
                    $recompile = new RecompileFrontendAssets(
                        $container->make($abstract),
                        $container->make(LocaleManager::class)
                    );
                    $recompile->flush();
                }
            );
```
Same logic, only js, css, not locale cache.

Reply:
> (ps this is code duplication, maybe all three locations need a trait?)

Yea I think a trait'd be nice for this, we should remember to do that

> I think keeping the clearCache method then makes sense, but only to be used by the LanguagePack.

and Locales extender